### ISSUE TITLE
The paging component lacks accessibility.

### ISSUE THREAD

**Issue by lafray** [vtex/shoreline#909]
# Bug Report

- [ ] I've read and understood the [Contributing guidelines](https://admin-ui.vercel.app/contributing/onboarding) and have done my best effort to follow them.
- [ ] I've searched for any related issues and avoided creating a duplicate issue.

## What happened

According to the [sources I researched](https://design-system.w3.org/components/pagination.html), pagination to be more accessible should be within 'nav' tags and have an arial-label with pagination. Neither of these items are present in the admin-ui pagination

## What I expected to happen

- Pagination component have `nav` tag and 
- Pagination component has an `arial-label` with `Pagination`.

## Reproducing

-  Inpect some Pagination component

## Attachments

- https://design-system.w3.org/components/pagination.html
- https://www.magentaa11y.com/checklist-web/pagination/
